src/Sparrow/Collections/RingBuffer.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+namespace Sparrow.Collections
```

Comment:
Better to move it to Sparrow.Server if this is not used in Raven.Client

Reply:
Botched merge, that is not used. removing.

Reply:
That is going to be used by RavenDB-12799 (https://issues.hibernatingrhinos.com/issue/RavenDB-12799) on `Sparrow`

Reply:
What is this going to be used _in sparrow_ ?

Reply:
Okay, this is used by the logging infrastructure, which we want to keep in sparrow